starknet_py/net/account/account_client.py

```diff
@@ -55,34 +55,39 @@ def __init__(
         client: Client,
         signer: Optional[BaseSigner] = None,
         key_pair: Optional[KeyPair] = None,
+        chain: Optional[StarknetChainId] = None,
     ):
         """
         :param address: Address of the account contract
         :param client: Instance of GatewayClient which will be used to add transactions
         :param signer: Custom signer to be used by AccountClient.
-                       If none is provieded, default
+                       If none is provided, default
                        :py:class:`starknet_py.net.signer.stark_curve_signer.StarkCurveSigner` is used.
         :param key_pair: Key pair that will be used to create a default `Signer`
+        :param chain: ChainId of the chain used by the client
         """
         # pylint: disable=too-many-arguments
         if signer is None and key_pair is None:
             raise ValueError(
                 "Either a signer or a key_pair must be provied in AccountClient constructor"
             )
 
-        chain = chain_from_network(net=client.net, chain=client.chain)
+        if chain is None and signer is None:
+            raise ValueError("One of chain or signer must be provided")
 
-        if chain is None and client is None:
-            raise ValueError("One of chain or client must be provided")
         self.address = parse_address(address)
         self.client = client
-        self.signer = signer or StarkCurveSigner(
-            account_address=self.address, key_pair=key_pair, chain_id=self.client.chain
-        )
+
+        if signer is None:
+            chain = chain_from_network(net=client.net, chain=chain)
+            signer = StarkCurveSigner(
+                account_address=self.address, key_pair=key_pair, chain_id=chain
+            )
+        self.signer = signer
 
     @property
     def chain(self) -> StarknetChainId:
-        return self.client.chain
+        return self.signer.chain_id
```

Comment:
I wonder if it makes sense to expose this property here at all if it's going through signer anyway.